crates/ra_ide/src/syntax_highlighting.rs

```diff
@@ -498,8 +519,8 @@ fn highlight_element(
         }
         NAME_REF => {
             let name_ref = element.into_node().and_then(ast::NameRef::cast).unwrap();
-            match classify_name_ref(sema, &name_ref) {
-                Some(name_kind) => match name_kind {
+            match (classify_name_ref(sema, &name_ref), is_possibly_unsafe(&name_ref)) {
```

Comment:
This match doesn't seem to inspect the `is_possibly_unsafe(&name_ref)`, so maybe just put it in a variable before the match?

Reply:
Done